### ISSUE TITLE
bug: duplicates in aws_identitystore_users and aws_identitystore_groups

### ISSUE THREAD

**Issue by chr-b** [cloudquery/cloudquery#15445]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

CloudQuery AWS plugin is duplicating results from the AWS Identity Center resources.
The data collection process for only those few tables takes about 13 hours. Another AWS organization of similar size completes in 2 hours (for a lot more tables).

Here are the results of a sync process (the numbers for distinct correspond to what the AWS CLI is returning):
```
select count(distinct user_id) from aws_identitystore_users;
count|666
select count(distinct _cq_id) from aws_identitystore_users;
count|673860
select count(*) from aws_identitystore_users;
count|673860

select count(distinct group_id) from aws_identitystore_groups;
count|853
select count(distinct _cq_id) from aws_identitystore_groups;
count|863236
select count(*) from aws_identitystore_groups;
count|863236
```

A consequence of this is that the number of rows in the group membership table has exploded:
```
select count(*) from aws_identitystore_group_memberships ;
count|7819967
```

### Expected Behavior

For `aws_identitystore_users` : number of rows is equal to the number of distinct `user_id`s
For `aws_identitystore_groups` : number of rows is equal to the number of distinct `group_id`s

(when using `write_mode: "overwrite-delete-stale"`)

### CloudQuery (redacted) config

```yaml
kind: source
spec:
  name: "aws"
  path: "cloudquery/aws"
  version: "v22.19.2"
  tables: ["aws_identitystore_groups", "aws_identitystore_group_memberships", "aws_identitystore_users", "aws_ssoadmin_instances", "aws_ssoadmin_permission_sets", "aws_ssoadmin_permission_set_account_assignments"]
  skip_dependent_tables: true
  destinations: ["aws_pq"]
  spec:
    org:
      admin_account:
        local_profile: "my-role"
      member_role_name: My_Role
    regions: ['*']
    aws_debug: true

kind: destination
spec:
  name: "aws_pq"
  registry: "github"
  path: "cloudquery/postgresql"
  version: "v7.0.1"
  write_mode: "overwrite-delete-stale"
  migrate_mode: "forced"
  spec:
    connection_string: "postgresql://USER:PASS@dbserver/aws?sslmode=prefer"
```

### CloudQuery (redacted) logs

```
2023-11-23T08:34:26Z INF Loading spec(s) args=["./"] module=cli
2023-11-23T08:34:28Z INF Plugin server listening address=/tmp/cq-LtwCehQubVfwjdan.sock module=cli
2023-11-23T08:34:29Z INF Plugin server listening address=/tmp/cq-DbIMhuZAcpWZHGgz.sock module=cli
2023-11-23T08:34:31Z INF Plugin server listening address=/tmp/cq-nzEOvQXKELcMDfaS.sock module=cli
2023-11-23T08:34:32Z INF started call grpc.component=server grpc.method=GetVersions grpc.method_type=unary grpc.service=cloudquery.discovery.v1.Discovery grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=0.014 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z INF finished call grpc.code=OK grpc.component=server grpc.method=GetVersions grpc.method_type=unary grpc.service=cloudquery.discovery.v1.Discovery grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=0.039 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z INF started call grpc.component=server grpc.method=GetVersions grpc.method_type=unary grpc.service=cloudquery.discovery.v1.Discovery grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=0.013 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z INF finished call grpc.code=OK grpc.component=server grpc.method=GetVersions grpc.method_type=unary grpc.service=cloudquery.discovery.v1.Discovery grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=0.04 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z INF started call grpc.component=server grpc.method=GetVersions grpc.method_type=unary grpc.service=cloudquery.discovery.v1.Discovery grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=0.012 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z INF finished call grpc.code=OK grpc.component=server grpc.method=GetVersions grpc.method_type=unary grpc.service=cloudquery.discovery.v1.Discovery grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=0.043 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z INF Start sync destinations=["assetdb_pq (postgresql@v7.0.1)","file_json (file@v3.4.11)"] module=cli source="aws_collector (aws@v22.19.2)" sync_time=2023-11-23T08:34:32Z
2023-11-23T08:34:32Z INF started call grpc.component=server grpc.method=Init grpc.method_type=unary grpc.service=cloudquery.plugin.v3.Plugin grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=0.012 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z INF Initializing postgresql destination module=pg-dest pgx_log_level=error
2023-11-23T08:34:32Z INF finished call grpc.code=OK grpc.component=server grpc.method=Init grpc.method_type=unary grpc.service=cloudquery.plugin.v3.Plugin grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=50.07 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z INF started call grpc.component=server grpc.method=Init grpc.method_type=unary grpc.service=cloudquery.plugin.v3.Plugin grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=0.011 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z INF finished call grpc.code=OK grpc.component=server grpc.method=Init grpc.method_type=unary grpc.service=cloudquery.plugin.v3.Plugin grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=0.2 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z INF started call grpc.component=server grpc.method=Init grpc.method_type=unary grpc.service=cloudquery.plugin.v3.Plugin grpc.start_time=2023-11-23T08:34:32Z grpc.time_ms=0.015 module=cli peer.address=@ protocol=grpc
2023-11-23T08:34:32Z ERR failed JSON schema validation for spec error="jsonschema: '/org/admin_account' does not validate with https://github.com/cloudquery/cloudquery/plugins/source/aws/client/spec/spec#/$ref/properties/org/oneOf/0/$ref/properties/admin_account/oneOf/0/$ref/required: missing properties: 'id'" module=aws-src
2023-11-23T08:34:44Z INF All regions specified in `cloudquery.yml`. Assuming all regions account=01234567890 module=aws-src
[...]
2023-11-23T08:41:18Z INF top level table resolver started client=1234567890123:us-east-1 module=aws-src table=aws_identitystore_groups
2023-11-23T08:41:19Z ERR table resolver finished with error error="operation error identitystore: ListGroups, 1 validation error(s) found.\n- missing required field, ListGroupsInput.IdentityStoreId.\n" client=1234567890123:us-east-1 module=aws-src table=aws_identitystore_groups
2023-11-23T08:41:19Z INF table sync finished client=1234567890123:us-east-1 errors=1 module=aws-src resources=0 table=aws_identitystore_groups
2023-11-23T08:41:19Z INF table sync finished client=1234567890123:us-east-1 errors=0 module=aws-src resources=0 table=aws_identitystore_group_memberships
2023-11-23T21:35:35Z INF top level table resolver started client=1234567890123:us-east-1 module=aws-src table=aws_identitystore_users
2023-11-23T21:35:35Z ERR table resolver finished with error error="operation error identitystore: ListUsers, 1 validation error(s) found.\n- missing required field, ListUsersInput.IdentityStoreId.\n" client=1234567890123:us-east-1 module=aws-src table=aws_identitystore_users
2023-11-23T21:35:35Z INF table sync finished client=1234567890123:us-east-1 errors=1 module=aws-src resources=0 table=aws_identitystore_users
```

### CloudQuery version

v3.28.1

### Additional Context

AWS plugin v22.19.2
Postgres plugin v7.0.1

### Pull request (optional)

- [ ] I can submit a pull request

**Comment by bbernays** [comment]
This is the expected behavior that the Users and Groups would be duplicated for each account that is synced. In the upcoming version `23.0.0`, we have made it more explicit by adding the `request_account_id` to enable users to identify where the request came from. If you only want to sync that data once you can update your config to only sync from a single account rather than all accounts in the `org`

**Comment by chr-b** [comment]
Hi @bbernays ,
Thanks for the feedback.

I am a little bit puzzled though. The identity store instance that is affected by this was created as part of IAM Identity Center in conjunction with AWS Organizations. Hence this is an organization level ID store. It exists in the organization management account (see also [AWS documentation](https://docs.aws.amazon.com/singlesignon/latest/userguide/organization-instances-identity-center.html)).

Hence, I had the assumption that this identity store will only be synched once, from within the organization management account.


Please consider that AWS recently extended SSO/ID Store: it is now possible to also have account level ID stores (in contrast to an organization level ID store).
From the [AWS documentation](https://docs.aws.amazon.com/singlesignon/latest/userguide/account-instances-identity-center.html): _Account instances are bound to a single AWS account and are used only to manage user and group access for supported AWS managed applications in the same account and AWS Region._ 

There is an accompanying blog post about this: https://aws.amazon.com/blogs/security/how-to-use-multiple-instances-of-aws-iam-identity-center/

**Comment by bbernays** [comment]
Hi @chr-b -

I believe that the organizational vs account level IAM Identity Center indicates the level at which it can federate access. The Org Instance can give access to any account in the organization in the account. 

When you run `aws sso-admin list-instances` from a non management account it will return the account level IAM Identity Instance as well as the Organizational level instance. 

In the interest of ensuring users have as complete of a picture of their environment CloudQuery will return all of the data possible to the user. 

Just so you know CloudQuery just added support for multiple instances of IAM Identity Center yesterday and it will be published in `v23.0.0`
